core/transaction.go

```diff
@@ -428,20 +431,45 @@ func ParseBlockVersion(protocolVersion string) (*semver.Version, error) {
 func eventCommitment(receipts []*TransactionReceipt) (*felt.Felt, error) {
 	var commitment *felt.Felt
 	return commitment, trie.RunOnTempTrie(commitmentTrieHeight, func(trie *trie.Trie) error {
-		count := uint64(0)
-		for _, receipt := range receipts {
-			for _, event := range receipt.Events {
-				eventHash := crypto.PedersenArray(
-					event.From,
-					crypto.PedersenArray(event.Keys...),
-					crypto.PedersenArray(event.Data...),
-				)
-
-				if _, err := trie.Put(new(felt.Felt).SetUint64(count), eventHash); err != nil {
-					return err
-				}
-				count++
+		eventCount := uint64(0)
+		numWorkers := runtime.GOMAXPROCS(0)
+		receiptPerWorker := len(receipts) / numWorkers
+		if receiptPerWorker == 0 {
+			receiptPerWorker = 1
+		}
+		workerPool := pool.New().WithErrors().WithMaxGoroutines(numWorkers)
+		var trieMutex sync.Mutex
+
+		for receiptIdx := range receipts {
+			if receiptIdx%receiptPerWorker == 0 {
+				curReceiptIdx := receiptIdx
+				curEventIdx := eventCount
+
+				workerPool.Go(func() error {
+					for idx := curReceiptIdx; idx < curReceiptIdx+receiptPerWorker && idx < len(receipts); idx++ {
+						for _, event := range receipts[idx].Events {
```

Comment:
Can we please use reslice so it would be easier to read i.e.:
```
maxIndex := curReceiptIdx + receiptPerWorker
if maxIndex > len(receipts) {
   maxIndex = len(receipts)
}
receipts := receipts[curReceiptIdx:maxIndex]

for _, receipt := range receipts {
	for _, event := range receipt.Events {
```

Reply:
Yeap, thanks. Done